docs/components/toaster.md

```diff
@@ -64,192 +57,207 @@ If a global property conflicts with a component’s own property, the component'
+  message: 'Detailed message. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.' })"
+>
+  Open Toaster
+</KButton>
 ```
 
-### timeout
+### appearance
 
-The default timeout, in milliseconds, is `5000` (5 seconds) however you can change this to by passing an override argument.
+Depending on the nature of notification, you might want to use different appearances. KToaster supports these 5 appearances:
+* `info` (default)
+* `success`
+* `danger`
+* `warning`
+* `system`
+
+<div class="horizontal-container">
+  <KButton @click="$toaster.open({ title: 'Info', appearance: 'info' })">
+    <InfoIcon />
+    Info
+  </KButton>
+  <KButton @click="$toaster.open({ title: 'Success', appearance: 'success' })">
+    <CheckCircleIcon />
+    Success
+  </KButton>
+  <KButton
+    @click="$toaster.open({ title: 'Danger', appearance: 'danger' })"
+    appearance="danger"
+  >
+    <ClearIcon />
+    Danger
+  </KButton>
+  <KButton @click="$toaster.open({ title: 'Warning', appearance: 'warning' })">
+    <WarningIcon />
+    Warning
+  </KButton>
+  <KButton
+    @click="$toaster.open({ title: 'System', appearance: 'system' })"
+    appearance="secondary"
+  >
+    <KongIcon />
+    System
+  </KButton>
+</div>
 
-- `timeoutMilliseconds`
+```ts
+$toaster.open({ title: 'Warning', appearance: 'warning' })
+```
 
-<KButton :disabled="timeLeft <= 3" @click="openNotificationElapse({timeoutMilliseconds: 3000, 'appearance': 'success', 'message': `This toaster has a 3 second timeout`})">{{timeLeft > 3 ? 'Open Toaster' : `Closing in ${timeLeft} seconds` }}</KButton>
+### timeoutMilliseconds
 
-```html
+The default timeout, in milliseconds, is `5000` (5 seconds) however you can change this to by passing the `timeoutMilliseconds` property.
+
+<KButton :disabled="timeLeft <= 3" @click="openNotificationElapse({ timeoutMilliseconds: 3000, appearance: 'success', title: 'Notification', message: `This toaster has a 3 second timeout` })">
+  {{timeLeft > 3 ? 'Open Toaster' : `Closing in ${timeLeft} seconds` }}
+</KButton>
+
+```vue
 <template>
   <KButton @click="openNotification(toasterOptions)">Open Toaster</KButton>
 </template>
 
-<script lang="ts">
-import { defineComponent } from 'vue'
-
-export default defineComponent({
-  data() {
-    return {
-      toasterOptions: {
-        appearance: 'success',
-        timeoutMilliseconds: 3000,
-        message: 'This toaster has a 3 second timeout'
-      },
-    }
-  },
-  methods: {
-    openNotification(options: Record<string, any> | string) {
-      this.$toaster.open(options)
-    }
-  }
-})
+<script setup lang="ts">
+import type { Toast } from '@kong/kongponents'
+
+const toasterOptions: Toast = {
+  appearance: 'success',
+  timeoutMilliseconds: 3000,
+  title: 'Notification',
+  message: 'This toaster has a 3 second timeout'
+}
+
+const openNotification = (options: Toast | string) => {
+  $toaster.open(options)
```

Comment:
I don't think this example works when using the Composition API? Same comment for all instances.

Reply:
I think it would need to be something like this, showing how you can utilize via composable:

```typescript
import { ToastManager } from '@kong/kongponents'
import type { ToasterNotification } from '@kong/kongponents'

// Initialize the ToastManager as a singleton so a new instance isn't created each time
const toaster = new ToastManager()

export default function useToaster() {
  const notify = async (notification: Partial<ToasterNotification>): Promise<void> => {
    toaster.open({
      ...notification,
    })
  }

  return {
    notify,
  }
}
```

Reply:
It works and it is how we have it in script section of this file. This example refers to `timeoutMilliseconds` argument.

Reply:
In this file yes because it's using the Options API right?

Reply:
No, it's using script setup with `getCurrentInstance` (screenshot). Ultimately, I think you are right and this component docs need to be updated to feature composable approach - I just don't really want to do it in this PR - how about we do it in `beta` when we refactor it not to create a second app instance?

![Screenshot 2024-02-16 at 11 07 07 AM](https://github.com/Kong/kongponents/assets/36751160/c5c4365f-c538-4a72-a6e0-d23763761a08)

Reply:
works for me 👍🏼